includes/REST_API/Products_Controller.php

```diff
@@ -164,34 +166,56 @@ public function get_items( $request ) {
 		 * @var string $search_term
 		 */
 		$search_term = ! empty( $request['search'] ) ? $request['search'] : '';
-		
+
 		/**
 		 * Request context.
 		 *
 		 * @var string $orderby
 		 */
 		$orderby = ! empty( $request['orderby'] ) ? $request['orderby'] : 'date';
 
+		/**
+		 * Request context.
+		 *
+		 * @var int $page
+		 */
+		$page = ! empty( $request['page'] ) ? $request['page'] : 1;
+
+		/**
+		 * Request context.
+		 *
+		 * @var int $per_page
+		 */
+		$per_page = ! empty( $request['per_page'] ) ? $request['per_page'] : 100;
+
 		/**
 		 * Request context.
 		 *
 		 * @var string $order
 		 */
 		$order = ! empty( $request['order'] ) ? $request['order'] : 'desc';
-		
-		
-		$query_result = $query->get_search( $search_term, $orderby, $order );
+
+		$query_result = $query->get_search( $search_term, $page, $per_page, $orderby, $order );
 		if ( is_wp_error( $query_result ) ) {
 			return $query_result;
 		}
 
 		$products = [];
-		foreach ( $query_result as $product ) {
+		foreach ( $query_result['products'] as $product ) {
 			$data       = $this->prepare_item_for_response( $product, $request );
 			$products[] = $this->prepare_response_for_collection( $data );
 		}
 
-		return rest_ensure_response( $products );
+		/**
+		 * Response object.
+		 *
+		 * @var WP_REST_Response $response
+		 */
+		$response = rest_ensure_response( $products );
+
+		$response->header( 'X-WP-HAS-NEXT-PAGE', (string) $query_result['has_next_page'] );
```

Comment:
```suggestion
		$response->header( 'X-WP-HasNextPage', (string) $query_result['has_next_page'] );
```

This is more in line with other headers (e.g. `X-WP-TotalPages`, `X-WP-Upload-Attachment-ID`, `X-WP-Nonce`)